Packages: install natives@1.1.6 - fix gulp & graceful-fs

fujin opened the pull request:
* I discovered this with latest node, however was able to reproduce with last
  version of Carbon.

```
> betaflight-configurator@10.5.0 start /Users/aj/src/betaflight-configurator
> node node_modules/gulp/bin/gulp.js debug
fs.js:25
'use strict';
^
ReferenceError: internalBinding is not defined
    at fs.js:25:1
    at req_ (/Users/aj/src/betaflight-configurator/node_modules/natives/index.js:137:5)
    at Object.req [as require] (/Users/aj/src/betaflight-configurator/node_modules/natives/index.js:54:10)
    at Object.<anonymous> (/Users/aj/src/betaflight-configurator/node_modules/rpm-builder/node_modules/graceful-fs/fs.js:1:99)
    at Module._compile (internal/modules/cjs/loader.js:722:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:733:10)
    at Module.load (internal/modules/cjs/loader.js:620:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:560:12)
    at Function.Module._load (internal/modules/cjs/loader.js:552:3)
    at Module.require (internal/modules/cjs/loader.js:658:17)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! betaflight-configurator@10.5.0 start: `node node_modules/gulp/bin/gulp.js debug`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the betaflight-configurator@10.5.0 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/aj/.npm/_logs/2019-01-01T19_34_17_428Z-debug.log
```

ref: https://github.com/gulpjs/gulp/issues/2246

McGiverGim commented:
The appdmg is a dependency only for mac, you can see it in the same file, at the bottom. Maybe is not working?

fujin commented:
@McGiverGim Will investigate.

fujin commented:
The platform-dependent-modules config was missing a caret in the version specification. Testing..

fujin commented:
I think this *should* be fixed now. I added the gulp-appdmg to the `optionalDependencies` package.json section, which caused `package-lock.json` to update when installed on macOS, with the `platform-dependent-modules` called during _postinstall.

Noticed a lot of the `"dev": true` are being removed from the lock - not sure on that one. It also added the integrity information around the optional dependency (as I installed on a mac, I guess.)

McGiverGim commented:
If I'm not wrong, mac users must generate the lock file and not use the one uploaded to git or upload the generated.

There was a discussion about this and I think remember that no solution was found compatible with all the platforms. Maybe @mikeller can confirm this, I think he made some tests under mac.

fujin commented on package.json:
```diff
@@ -39,6 +39,7 @@
     "i18next-xhr-backend": "^1.5.1",
     "lru_map": "^0.3.3",
     "marked": "^0.5.2",
+    "natives": "^1.1.6",
```
@mikeller It fixes the stacktrace in the description (https://github.com/gulpjs/gulp/issues/2246)

McGiverGim replied:
The schedule of  Node is a little strange. They release a version each 6 months (april and october) and the older, if covered by the LTS, then it has18 months of LTS and then 12 months of maintenance.
Seeing the graph it seems version 11 will be a short version, and will end soon:
![schedule](https://user-images.githubusercontent.com/2673520/50816343-8ef7da80-1320-11e9-9a73-5e26f8e6be68.png)

Version 12 will be a long version, and released october of this year, so we can ignore version 11, but I'm not too sure version 12 will fix these problems. If it has the same problems, then we can go with versions 11 and 12 because the problems and solutions will be the same.

By other side, it seems to me is that the Node 11 has a bug, and the natives is simply installing some version in advance of the actual version included in our Node, that seems to fix the problem, so I hope version 12 maybe can fix it. Do you understand the same than me?

This seems only affect Mac. Is that right? Maybe then, can it be added into `platformDependentModules` to limit the effect?

mikeller replied:
@McGiverGim: Having to add `natives` as a direct dependency to make things work  in node 11 is ugly at least, but I'd much rather consider this a bug, so the hope is that this will be fixed at some stage.

fujin replied:
I was able to reproduce https://github.com/gulpjs/gulp/issues/2246 in our repository with the last release of Carbon (8.x) as well, however this could be due to a polluted local npm_modules directory or NPM cache from originally trying it with 11.x.

I'll try repro again. Sorry.

Docteh commented:
Is this still potentially an issue? I recently updated my macBookPro and had no trouble, even after updating nodejs to `v12.7.0`
